components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -1065,11 +1066,23 @@ private void handleTokenBinding(OAuth2AccessTokenReqDTO tokenReqDTO, String gran
                 throw new IdentityOAuth2ClientException(OAuth2ErrorCodes.INVALID_REQUEST,
                         "TLS certificate not found in the request.");
             }
+            if (OAuth2Constants.TokenBinderType.CLIENT_INSTANCE.equals(tokenBinder.getBindingType())) {
```

Comment:
Having specific checks for implementation in the core is not a good practice. What is the reason for these checks?

Reply:
If the client instance has not sent any value, we treat as no binding type (as the binding is set to null by default)